### PR TITLE
Create client service

### PR DESCRIPTION
**Pull request by RtAnto**
# Description

Added the client service, client service repo, and client service UT

## Type of change

- [X ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ X] Create Client
- [ X] Get Client
- [X ] Update Client
- [ X] Delete Client

# Notes

Unit testing could be expanded upon for handling exceptions

# Pivotal Tracker Link:

N/a

### CODE REVIEW

File: Backend/DataAccessLayer/Repositories/ClientRepository.cs [hunk 1]
```diff
@@ -10,14 +11,56 @@ namespace DataAccessLayer.Repositories
...
-                _db.Clients.Add(client);
-                return _db.SaveChanges();
+            _db.Entry(client).State = EntityState.Modified;
+            return client;
+
+        }
+        
+        public bool isClientDisabled(DatabaseContext _db, Client client)
+        {
+            if (client.Disabled) {
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/ClientRepository.cs]:
```suggestion
            var result = GetClient(_db, client.Id);
            if (result.Disabled) {
```

File: Backend/DataAccessLayer/Repositories/ClientRepository.cs [hunk 2]
```diff
@@ -10,14 +11,56 @@ namespace DataAccessLayer.Repositories
...
-            using (var _db = new DatabaseContext())
-            {
-                _db.Clients.Add(client);
-                return _db.SaveChanges();
+            _db.Entry(client).State = EntityState.Modified;
+            return client;
+
+        }
+        
+        public bool isClientDisabled(DatabaseContext _db, Client client)
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/ClientRepository.cs]:
```suggestion
        public bool IsClientDisabled(DatabaseContext _db, Client client)
```

File: Backend/ServiceLayer/Services/IClientService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ServiceLayer.Services
+{
+    public interface IClientService
+    {
+
```
Comment (Luke-Clements) [inline Backend/ServiceLayer/Services/IClientService.cs]: Need to add the CRUD methods signatures here

File: Backend/DataAccessLayer/Repositories/ClientRepository.cs [hunk 4]
```diff
@@ -10,14 +11,56 @@ namespace DataAccessLayer.Repositories
...
     public class ClientRepository
     {
-        public int CreateClient(Client client)
+        public Client CreateClient(DatabaseContext _db, Client client)
         {
+            _db.Entry(client).State = EntityState.Added;
+            return client;
+        }
+
+        public Client DeleteClient(DatabaseContext _db, Guid Id) {
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/ClientRepository.cs]:
```suggestion
        public Client DeleteClient(DatabaseContext _db, Guid Id) 
        {
```

File: Backend/DataAccessLayer/Repositories/ClientRepository.cs [hunk 5]
```diff
@@ -10,14 +11,56 @@ namespace DataAccessLayer.Repositories
...
+                .Where(c => c.Id == Id)
+                .FirstOrDefault<Client>();
+            if (client == null)
+                return null;
+            _db.Entry(client).State = EntityState.Deleted;
+            return client;
+
+        }
+
+        public Client GetClient(DatabaseContext _db, Guid Id) {
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/ClientRepository.cs]:
```suggestion
        public Client GetClient(DatabaseContext _db, Guid Id) 
        {
```

File: Backend/DataAccessLayer/Repositories/ClientRepository.cs [hunk 6]
```diff
@@ -10,14 +11,56 @@ namespace DataAccessLayer.Repositories
...
+            var client = _db.Clients
+                .Where(c => c.Id == Id)
+                .FirstOrDefault<Client>();
+            return client;
+        }
+
+        public Client UpdateClient(DatabaseContext _db, Client client) {
```
Comment (Luke-Clements) [inline Backend/DataAccessLayer/Repositories/ClientRepository.cs]:
```suggestion
        public Client UpdateClient(DatabaseContext _db, Client client) 
        {
```

**Review by ChrisMeyer7088 — APPROVED** [review]
Thanks for getting this out

**Pull request merged**